File: everyDay/2347.最好的扑克手牌.cpp
/*
 * @lc app=leetcode.cn id=2347 lang=cpp
 *
 * [2347] 最好的扑克手牌
 */

// @lc code=start
class Solution {
public:
    string bestHand(vector<int>& ranks, vector<char>& suits) {
        string ans;
        int f1 = 0, f2 = 0, f3 = 0;
        char la = 'A';
        for (auto &it : suits) {
            if (la != 'A' && la != it) {
                f1 = 1;
                break;
                }
            la = it;
        }
        if (!f1) return "Flush";
        unordered_map<int, int> M;
        for (auto &it : ranks) {
            M[it]++;
        }
        for (auto [x, y] : M) {
            if (y >= 3) {
                f2 = 1;
                break;
            }
            if (y == 2) {
                f3 = 1;
            }
        }
        if (f2) return "Three of a Kind";
        if (f3) return "Pair";
        return "High Card"; 
    }
};
// @lc code=end

File: everyDay/2347.最好的扑克手牌.go
/*
 * @lc app=leetcode.cn id=2347 lang=golang
 *
 * [2347] 最好的扑克手牌
 */

// @lc code=start
func bestHand(ranks []int, suits []byte) string {
    if bytes.Count(suits, suits[:1]) == 5 {
        return "Flush"
    }
    cnt, pair := map[int]int{}, false
    for _, r := range ranks {
        cnt[r]++
        if cnt[r] == 3 {
            return "Three of a Kind"
        }
        if cnt[r] == 2 {
            pair = true
        }
    }
    if pair {
        return "Pair"
    }
    return "High Card"
}
// @lc code=end

